arkouda/categorical.py

```diff
@@ -73,9 +73,7 @@ def __init__(self, values, **kwargs) -> None:
                                  "Strings not yet supported"))
             g = GroupBy(values)
             self.categories = g.unique_keys
-            self.codes = zeros(values.size, dtype=np.int64)
-            self.codes[cast(pdarray, g.permutation)] = \
-                                g.broadcast(arange(self.categories.size))
+            self.codes = g.broadcast(arange(self.categories.size), permute=True)
```

Comment:
We don't need the `permute=True` here since this is the default for broadcast now, right? 

It's probably good to be explicit in case the default does get changed back for whatever reason. No change requested